src/github.com/getlantern/fronted/direct.go

```diff
@@ -136,6 +137,8 @@ func (d *direct) Do(req *http.Request) (*http.Response, error) {
 // Dial persistently dials masquerades until one succeeds.
 func (d *direct) Dial(network, addr string) (net.Conn, error) {
 	gotFirst := false
+	candidateCh := <-candidateChCh
+	candidateChCh <- candidateCh
```

Comment:
BTW, the problem with putting candidateCh back on the candidateChCh is that you'll end up with two versions of candidateCh on candidateChCh after Configure() is called a second time (because candidateChCh is sized to 1).  After a 3rd call to Configure(), either Configure() or Dial() will get deadlocked because the channel is full at that point.

You could do this in Dial to resolve that:

```golang
select {
    case candidateChCh <- candidateCh:
        // okay
    default:
        // discard
}
```

If you make candidateChCh a blocking (size 0) channel, that also addresses the multiple versions of candidateCh issue.  Still, I think using an atomic.Value is cleaner.